rudder-core/src/main/scala/com/normation/rudder/reports/execution/ReportsExecutionRepositoryImpl.scala

```diff
@@ -95,14 +95,29 @@ case class RoReportsExecutionRepositoryImpl (
         // to interpolate the actual value of in - sql""" """ is more
         // than just interpolation. Need to check perfomances.
         (for {
-          runs <- sql"""select distinct on (r.nodeid)
-                         r.nodeid, r.date, r.nodeconfigid, r.complete, r.insertionid
-                       , c.nodeid, c.nodeconfigid, c.begindate, c.enddate, c.configuration
-                       from  reportsexecution r
-                       left outer join nodeconfigurations c
-                         on r.nodeId = c.nodeid and r.nodeconfigid = c.nodeconfigid
-                       where r.complete = true and r.nodeid in (${nodes : nodes.type})
-                       order by r.nodeid, r.insertionid desc
+          // Here to make the query faster, we distinct only on the reportexecution to get the last run
+          // but we need to get the matching last entry on nodeconfigurations.
+          // I didn't find any better solution than doing a distinct on the table
+          // However, I am surprised that we do a left outer join:
+          // we cannot do a search reports on a node that we don't know of
+          // so we will always have a matching nodeconfigurations (except after migrating from 3.x to 4.x)
```

Comment:
no, we can have agent run without node configurations: 
- before first generation, 
- for deleted nodes (they are still sending runs)
- certainly for a lot of other cases. 
Perhaps amend the comment, I'm ok witht the query so you can self merge afterward.